src/coreclr/tools/Common/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -200,7 +200,26 @@ private EcmaModule AddModule(string filePath, string expectedSimpleName, bool us
                     throw new NotSupportedException($"Error: C++/CLI is not supported: '{filePath}'");
 #endif
 
-                    pdbReader = PortablePdbSymbolReader.TryOpenEmbedded(peReader, GetMetadataStringDecoder()) ?? OpenAssociatedSymbolFile(filePath, peReader);
+                    try
+                    {
+                        pdbReader = PortablePdbSymbolReader.TryOpenEmbedded(peReader, GetMetadataStringDecoder())
+                                    ?? OpenAssociatedSymbolFile(filePath, peReader);
+                    }
+                    catch (BadImageFormatException e)
```

Comment:
This is placed in a very broad spot. A lot of code is going to run and could throw a BadImageFormat (S.R.Metadata will throw it for any sort of corruption).

Can you instead add a e.g. `SafeReadDebugDirectory(this PEReader peReader)` extension method that simply checks the directory size before calling the API and returns `ImmutableArray<DebugDirectoryEntry>.Empty` if it doesn't make sense?

But doing Console.WriteLine in arbitrary spots is not a great practice because we're multithreaded. I would honestly just swallow this.

Reply:
I did not consider the fact of multithreading. Thanks for your feedback Michal! I think this is a great suggestion, I'll get to work on it.